week06/completed_product_test.go

```diff
@@ -0,0 +1,38 @@
+package week06
+
+import (
+	"testing"
+)
+
+func TestCompletedProductIsValid(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name	string
+		emp 	Employee
+		product Product
+		err 	error
+	}{
+		{name: "validity of invalid product", emp: Employee(1), product: Product{Quantity: -1} , err: ErrInvalidQuantity(-1)},
+		{name: "validity of product built by invalid employee", emp: Employee(-1), product: Product{Quantity: 1}, err: ErrInvalidEmployee(-1)},
+		{name: "validity of valid product", emp: Employee(1), product: Product{Quantity: 1}, err: nil},
+	}
+	for _, tc := range testCases {
+
+		t.Run(tc.name, func(t *testing.T) {
+
+			_ = tc.product.Build(tc.emp)
```

Comment:
Never ignore an `error`. If there is a problem with the `Build` method and it returns an `error`, the rest of your tests may fail or panic as a result. Checking the error allows you to fail the test gracefully.